Source/player.h

```diff
@@ -418,11 +418,22 @@ struct PlayerStruct {
 			// Maximum achievable HP is approximately 1200. Diablo uses fixed point integers where the last 6 bits are
 			// fractional values. This means that we will never overflow HP values normally by doing this multiplication
 			// as the max value is representable in 17 bits and the multiplication result will be at most 23 bits
-			_pHPPer = _pHitPoints * 80 / _pMaxHP;
+			_pHPPer = clamp(_pHitPoints * 80 / _pMaxHP, 0, 80); // hp should never be greater than maxHP but just in case
 		}
 
 		return _pHPPer;
 	}
+
+	int UpdateManaPercentage()
+	{
+		if (_pMaxMana <= 0) {
+			_pManaPer = 0;

```

Comment:
I looked at the Devilution code and _pManaPer is not relevant so we can stript it from the struct and skip it in the saves. Would be nice to get that done to clean this up as well.

Reply:
I agree they don't need to be saved, vanilla recalculates them when setting up the game state (and on every frame) so that won't cause issues in old versions. I'd like to keep the variables in the struct though as they serve a useful purpose for the UI code. The HP and Mana values change relatively infrequently so there's no real need to redo this calculation 3 times each every frame. Ideally this is done when changing the value of `_Mana`/`_MaxMana`/`_Life`/`_MaxLife` and the draw routines can just rely on the `*Per` value being accurate

Reply:
Go for it :)